snapshots/src/error.rs

```diff
@@ -95,6 +97,12 @@ pub enum SnapshotError {
     RebuildStorages(String),
 }
 
+impl From<SendError<FileInfo>> for SnapshotError {
+    fn from(err: SendError<FileInfo>) -> Self {
+        Self::CrossbeamSend(SendError(err.into_inner().path))
+    }
+}
```

Comment:
Do we always have a FileInfo now? If yes, let's update the `CrossbeamSend` error to hold a FileInfo instead of a PathBuf.